addons/base/views.py

```diff
@@ -869,6 +869,14 @@ def addon_view_file(auth, node, file_node, version):
         args={'url': download_url.url}
     )
 
+    version_names = []
+    versions = file_node.versions.order_by('-created')
+    if versions.count():
+        for vers in versions:
+            file_version_thru = vers.get_basefilenode_version(file_node)
+            name = file_version_thru.version_name if file_version_thru else file_node.name
```

Comment:
With `version_name` blankable, is it possible that `name` here would be assigned as `''` despite `file_node.name` being populated?